### ISSUE TITLE
Allow for multi-tenancy

### ISSUE THREAD

**Issue by dominguez-aureax** [dominguez-aureax/aureax-web#12]

**Comment by dominguez-aureax** [comment]
There are two ways we can progress with Multi-tenancy, either using the Identity Platform with Google Cloud:
https://cloud.google.com/identity-platform/docs/multi-tenancy-authentication#email_link
or 
to set up different projects for each tenant
https://stackoverflow.com/questions/54150421/is-a-multi-tenancy-setup-feasible-with-firebase-and-firestore

I prefer the first option because it also opens up a lot of the Firebase API tools for multi-tenancy

**Comment by jsstallings** [comment]
I agree with option 1.  This is multi-tenancy for identity mgmt, does what you already have for persistence of data in firebase already have multi-tenancy as well, or is that still TBD?

**Comment by jsstallings** [comment]
I think it's both, the multi-tenancy from the IdM perspective, and the data model itself in firebase.  This short article kind of explains it, that we should have a data model that supports orgs as we talked about (much like Schools in his example), and that when we deploy a new customer org it creates data in the existing data structure, and provisions them from an IdM perspective for multi-tenant access.  

https://christianlydemann.com/setting-up-a-multi-tenant-application-with-firebase-graphql-and-angular/

**Comment by dominguez-aureax** [comment]
My idea would be to take away some of the logic used here as I believe that this approach is for one application supporting multiple tenants. I would approach it like so:

1. Each application only has reference to the authentication id for their unique company using the Google Identity Platform.
2. The application will only have majority access (I say majority because there is a possibility of it still needing a reference to some global values) to their own section of the database. For instance, the application will load in information for companyA, but there is no connection to companyB because I am planning to only point a collection 

Example:
```
const company= await firestore().collection('aureax').get()
```

the value 'aureax' will change between each unique application and since there is no need for multiple company variables, this takes away the hassle of routing. However, now that I write this I see that there may be a need for routing in the web version... so I may need to adapt some of my plans now. Thanks for the info!
